test/course_planner_web/controllers/event_controller_test.exs

```diff
@@ -0,0 +1,206 @@
+      conn = get conn, event_path(conn, :edit, event)
+      html_response(conn, 403)
+    end
+
+    @tag user_role: :coordinator
+    test "renders form for editing chosen event", %{conn: conn, event: event} do
+      conn = get conn, event_path(conn, :edit, event)
+      assert html_response(conn, 200) =~ event.name
+    end
+  end
+
+  describe "update event" do
+    @tag user_role: :student
+    test "student cannot update event", %{conn: conn, event: event} do
+      conn = put conn, event_path(conn, :update, event), event: @update_attrs
+      html_response(conn, 403)
+    end
+
+    @tag user_role: :teacher
+    test "teacher cannot update event", %{conn: conn, event: event} do
+      conn = put conn, event_path(conn, :update, event), event: @update_attrs
+      html_response(conn, 403)
+    end
+
+    @tag user_role: :volunteer
+    test "volunteer cannot update event", %{conn: conn, event: event} do
+      conn = put conn, event_path(conn, :update, event), event: @update_attrs
+      html_response(conn, 403)
+    end
+
+    @tag user_role: :coordinator
+    test "redirects when data is valid", %{conn: conn, event: event} do
+      conn = put conn, event_path(conn, :update, event), event: @update_attrs
+      assert redirected_to(conn) == event_path(conn, :show, event)
+
+      conn = get conn, event_path(conn, :show, event)
+      assert html_response(conn, 200) =~ @update_attrs.name
+    end
+
+    @tag user_role: :coordinator
+    test "renders errors when data is invalid", %{conn: conn, event: event} do
+      conn = put conn, event_path(conn, :update, event), event: @invalid_attrs
+      assert html_response(conn, 200) =~ event.name
+    end
+  end
+
+  describe "delete event" do
+    @tag user_role: :student
+    test "student cannot delete event", %{conn: conn, event: event} do
+      conn = delete conn, event_path(conn, :delete, event)
+      response(conn, 403)
+      conn = get conn, event_path(conn, :show, event)
+      response(conn, 200) =~ event.name
+    end
+
+    @tag user_role: :teacher
+    test "teacher cannot delete event", %{conn: conn, event: event} do
+      conn = delete conn, event_path(conn, :delete, event)
+      response(conn, 403)
+      conn = get conn, event_path(conn, :show, event)
+      response(conn, 200) =~ event.name
+    end
+
+    @tag user_role: :volunteer
+    test "volunteer cannot delete event", %{conn: conn, event: event} do
+      conn = delete conn, event_path(conn, :delete, event)
+      response(conn, 403)
+      conn = get conn, event_path(conn, :show, event)
+      response(conn, 200) =~ event.name
+    end
+
+    @tag user_role: :coordinator
+    test "deletes chosen event", %{conn: conn, event: event} do
+      conn = delete conn, event_path(conn, :delete, event)
+      assert redirected_to(conn) == event_path(conn, :index)
+      conn = get conn, event_path(conn, :show, event)
+      response(conn, 404)
+    end
+  end
```

Comment:
you do not test deletion of a non existing resource when user is not permitted. the point of testing it is to see which error will be visible. resource does not exist or action is not permitted.

Reply:
Good catch, I'll add that.